packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -2429,3 +2429,71 @@ contract LockedGoldGetTotalPendingWithdrawalsCount is LockedGoldTest {
     assertEq(lockedGold.getTotalPendingWithdrawalsCount(randomAddress), 0);
   }
 }
+
+contract getPendingWithdrawalsInBatch is LockedGoldTest {
+  uint256 value = 1000;
+
+  function setUp() public {
+    super.setUp();
+  }
+
+  function test_ShouldReturn0_WhenAccountHasNoPendingWithdrawals() public {
+    (uint256[] memory pendingWithdrawals, uint256[] memory timestamps) = lockedGold
+      .getPendingWithdrawalsInBatch(randomAddress, 0, 0);
+    assertEq(pendingWithdrawals.length, 0);
+    assertEq(timestamps.length, 0);
+  }
+
+  function test_ShouldReturnCorrectValue_WhenAccountHasPendingWithdrawals() public {
+    lockedGold.lock.value(value)();
+
+    lockedGold.unlock(value / 2);
+    lockedGold.unlock(value / 2);
+
+    (uint256[] memory pendingWithdrawals, uint256[] memory timestamps) = lockedGold
+      .getPendingWithdrawalsInBatch(caller, 0, 1);
+    assertEq(pendingWithdrawals.length, 2);
+    assertEq(timestamps.length, 2);
+    assertEq(pendingWithdrawals[0], value / 2);
+    assertEq(pendingWithdrawals[1], value / 2);
+  }
+
+  function test_ShouldReturnCorrectValue_WhenAccountHasFourPendingWithdrawals() public {
+    lockedGold.lock.value(value)();
+
+    lockedGold.unlock(value / 4 - 1);
+    lockedGold.unlock(value / 4 + 1);
+    lockedGold.unlock(value / 4 - 2);
+    lockedGold.unlock(value / 4 + 2);
+
+    (uint256[] memory pendingWithdrawals, uint256[] memory timestamps) = lockedGold
+      .getPendingWithdrawalsInBatch(caller, 0, 1);
+    assertEq(pendingWithdrawals.length, 2);
+    assertEq(timestamps.length, 2);
+    assertEq(pendingWithdrawals[0], value / 4 - 1);
+    assertEq(pendingWithdrawals[1], value / 4 + 1);
+    (pendingWithdrawals, ) = lockedGold.getPendingWithdrawalsInBatch(caller, 2, 3);
+    assertEq(pendingWithdrawals.length, 2);
+    assertEq(timestamps.length, 2);
+    assertEq(pendingWithdrawals[0], value / 4 - 2);
+    assertEq(pendingWithdrawals[1], value / 4 + 2);
+  }
+
+  function test_Reverts_WhenInvalidRangeProvided_WhenAccountHasPendingWithdrawals() public {
+    lockedGold.lock.value(value)();
+
+    lockedGold.unlock(value / 2);
+    lockedGold.unlock(value / 2);
+
+    vm.expectRevert("Invalid range");
+    (uint256[] memory pendingWithdrawals, uint256[] memory timestamps) = lockedGold
+      .getPendingWithdrawalsInBatch(caller, 0, 2);
+  }
+
+  function test_ShouldReturn0_WhenNonExistentAccount() public {
+    (uint256[] memory pendingWithdrawals, uint256[] memory timestamps) = lockedGold
+      .getPendingWithdrawalsInBatch(randomAddress, 0, 1);
```

Comment:
Slight inconsistency in the logic: a non-existing account will return 0 length arrays for whatever size batch we ask, an existing account will revert when asking for a range that is too large.

Would recommend either:
* not reverting if range is larger than array length (just return as much as possible)
* or reverting on accounts that don't have pending withdrawals.

Reply:
+1 for returning as much as possible.

Reply:
Since reverting would be backward incompatible we are now returning "as much as possible"